### PR TITLE
API endpoint for locations

### PR DESCRIPTION
**Pull request by martintomas**
I am adding two new API endpoints for location --> one for `index` and second for `show/detail`.

Only other thing which is probably worthy to mention is that `index` locations API endpoint allows to filter locations based on:
 - location_type
 - parent_id

I have also added LocationType values to Enum ;).

## Testing instructions

Please check swagger documentation, there are two new api endpoints for locations:
Screenshot 2022-03-21 at 11 42 21

```
{
  "data": {
    "id": "4debca7f-9251-403f-b893-ff0e78bd6af4",
    "type": "location",
    "attributes": {
      "name": "Hane, Lehner and Goyette",
      "location_type": "municipality"
    },
    "relationships": {
      "parent": {
        "data": {
          "id": "5331a4ba-ef75-4b7f-8cdd-8ec39e692bbd",
          "type": "location"
        }
      },
      "regions": {
        "data": [
          {
            "id": "b5debe16-a4ab-4811-ac48-3b480af1fc56",
            "type": "location"
          }
        ]
      }
    }
  }
}
```

## Tracking

https://vizzuality.atlassian.net/browse/LET-129

### CODE REVIEW

File: backend/app/models/location_type.rb
Comment (martintomas) [inline backend/app/models/location_type.rb:5]: I am renaming `state` to `country` so this keeps naming convention used at design

**Review by tsubik — APPROVED** [review]
Nice!

File: backend/app/models/location_type.rb
Comment (agnessa) [inline backend/app/models/location_type.rb:6]: do we then need to rename department to state?

Comment (martintomas) [reply backend/app/models/location_type.rb:6]: :D should I? I am fine with renaming `state` to `country` because it still makes sense for me, but in this case `department` still makes more sense to me then `state`. Also wiki and Santiago's data speak about these entities as departments. But if you wanna go for `state`, I will do it, no problem -- just please let me know your final word and I will tweak it eventually before merging this 🙏

Comment (agnessa) [reply backend/app/models/location_type.rb:6]: you're right, in the file it is called department, in the design state. Let's stick with department for now then.

**Pull request merged**
